### PR TITLE
Stats: Fix tooltip flash on new sites

### PR DESCRIPTION
**Pull request by a8ck3n**
Related to:

https://github.com/Automattic/red-team/issues/192

## Proposed Changes

Guard against the `isNewSite` value being `null`. When that happens and the site is less than one week old, the value goes from `null` (falesy) to `true` causing the tooltip to show for a few seconds and then disappear. It's a bit jarring.

This fix guards against the unexpected `null` in the component logic. Ideally, we'd fix the logic in the `connect()` function to only pass boolean values. I'll look into that next.

## Why are these changes being made?

To prevent the Settings cog's tooltip flashing onscreen then hiding on new sites.

## Testing Instructions

Somewhat difficult to test as we don't have an easy way to reset the inputs.

1. Check out this branch at commit `b975196bdf68301f610d77ef84aa6b57ddca7994`. This commit contains code to force the issue.
2. Test with local Jetpack/Calypso environment to see the issue in action. Option 1 here: PCYsg-Pp7-p2
3. Move to commit `22220ea1ee44a76dd51674d6e9c28d4e3a7ebecf`. (debug conditions & fix are present)
4. Confirm the tooltip is no longer popping in and out.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/blocks/stats-navigation/index.js [hunk 1]
```diff
@@ -167,6 +168,11 @@ class StatsNavigation extends Component {
 
 		// @TODO: Add loading status of modules settings to avoid toggling modules before they are loaded.
 
+		// The isNewSite value can be null so we need to guard against that.
+		// This is semantically incorrect as we expect a boolean value.
+		// TODO: Update logic in connect() function to send only true/false value.
+		const delayTooltipPresentation = isNewSite === null || isNewSite;
```
Comment (dognose24) [inline client/blocks/stats-navigation/index.js:174]: Nit: Could we use a more specific naming to describe the relation with the `isNewSite,` e.g., `showTooltipForOldSite`, to possibly reduce the annotations?
```suggestion
		const showTooltipForOldSite = isNewSite === false;
```

**Review by dognose24 — APPROVED** [review]
Works as described! Left a nit that is not a blocker.

File: client/blocks/stats-navigation/index.js (hunk 1 above)
Comment (a8ck3n) [reply client/blocks/stats-navigation/index.js:174]: The extra comments shouldn't be needed once I fix up the logic in the `connect()` function. I'll look to update the property name to better represent the updated logic too.

Comment (a8ck3n) [reply client/blocks/stats-navigation/index.js:174]: Addressed in https://github.com/Automattic/wp-calypso/pull/95509.

**Pull request merged**